build/package/scripts/build-typescript.sh

```diff
@@ -97,12 +111,17 @@ fi
 
 echo "Building ..."
 npm run build
+# Copying most build output before testing so
+# that additional modules which may be installed by testing 
+# is not included.
+# However copying package.json too early can confuse the tests.
 mkdir -p "${OUTPUT_DIR}"
-cp -r "${BUILD_DIR}" "${OUTPUT_DIR}/dist"
+echo "Copying ${BUILD_DIR} to ${OUTPUT_DIR} ..."
+"$CP" --target-directory  "${OUTPUT_DIR}" -r "${BUILD_DIR}" 
```

Comment:
Why are we not copying to `dist` any longer? I'm fine with this but should probably call this out as a breaking change since people need to adapt their `Dockerfile` ...

Reply:
Out of curiosity: can you explain why you use GNU cp with `--target-directory` here? What's the benefit?

Reply:
Without the removal of `dist` there were actually duplicate levels `dist/dist`.  I believe this was not intended in the first place see resolved conversation for build-typescript.sh at https://github.com/opendevstack/ods-pipeline/pull/371/files. But I actually wanted to discuss this with @henninggross  and @kuebler as well when I am starting to believe that it is the right thing to do (that was one reason this is a draft at the moment).

I come to the conclusion that we should depend on GNU `cp` as this is what is available under our target os. I see these benefits:
-  With `--target-directory` the command will fail if you are not using GNU cp, for example when testing stuff locally on a Mac. I also adjusted the script so you can define env variable GNU_CP with a different name (which use useful to use the `g` prefixed commands installed by `brew install coreutils`.
- There is ambiguity on whether an argument is meant to be a directory or file if the target does not exist yet. There appears to be a grey zone of behaviors associated with that on how different versions/flavors of `cp` work. I hope that `--target-directory` is more clear here.  

So overall I am hoping to make this simpler! 

Why would I even run the build script locally?
- I found that using task tests is slows development down a lot. So I am using a small script so I can test the build scripts against our project code directly. Since this script is meant as a starting point, I shared it as a gist under  https://gist.github.com/henrjk/4265bc1b4f6fe3f7cd9027b0bcc284f9